File: src/log/statement/SimpleStatement.java
package log.statement;

/**
 * Définit une proposition basée sur une information unique
 */
class SimpleStatement implements Statement {
    // ATTRIBUTS
    /** La chaîne représentant l'objet décrit par cette proposition*/
    private final String content;


    // CONSTRUCTEUR
    <T> SimpleStatement(T object) {
        content = object.toString();
    }


    // REQUETE
    @Override
    public String toString() {
        return content;
    }
}

File: src/main/statement/SimpleStatement.scala
package main.statement

/**
  * Définit une proposition basée sur une information unique
  */
class SimpleStatement private[statement](val `object`: T) extends Statement {
  /** La chaîne représentant l'objet décrit par cette proposition */
  private val content: String = `object`.toString

  // REQUETE
  override def toString: String = content
}